iceberg/openhouse/internalcatalog/src/main/java/com/linkedin/openhouse/internal/catalog/OpenHouseInternalTableOperations.java

```diff
@@ -224,7 +227,8 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
           InternalCatalogMetricsConstant.METADATA_UPDATE_LATENCY);
 
       houseTable = houseTableMapper.toHouseTable(updatedMetadata);
```

Comment:
could we change the method here to ` houseTable = houseTableMapper.toHouseTable(updatedMetadata, fileIO)`  to keep object immutability and use mapping logic to correctly set FileIO

Reply:
Lets discuss in https://github.com/linkedin/openhouse/pull/187#discussion_r1750715536